packages/local-signing-manager/src/types/index.ts

```diff
@@ -1,16 +1,9 @@
 /**
  * URI|mnemonic|hex representation of a private key
  */
-export type PrivateKey =
-  | {
-      uri: string;
-    }
-  | {
-      mnemonic: string;
-    }
-  | {
-      seed: string;
-    };
+export type PrivateKey<S = string> = ({ uri: S } | { mnemonic: S } | { seed: S }) & {
```

Comment:
This should not be generic:
```suggestion
export type PrivateKey = ({ uri: string } | { mnemonic: string } | { seed: string }) & {
```

Reply:
This would be a breaking change, and a user should not able to change the type of their argument.